app/src/main/java/tech/relaycorp/gateway/data/database/ParcelCollectionDao.kt

```diff
@@ -18,4 +21,19 @@ interface ParcelCollectionDao {
 
     @Query("SELECT * FROM ParcelCollection")
     suspend fun getAll(): List<ParcelCollection>
+
+    @Query(
+        """
+        SELECT * FROM ParcelCollection
+        WHERE recipientAddress = :recipientAddress 
+            AND senderAddress = :senderAddress 
+            AND messageId = :messageId
+        LIMIT 1
+        """
+    )
+    suspend fun get(
+        recipientAddress: MessageAddress,
+        senderAddress: PrivateMessageAddress,
+        messageId: MessageId
+    ): ParcelCollection?
```

Comment:
Since we're not interested in the `ParcelCollection` object, but we just want to know if a record exists, I'd suggest to just return a boolean. For example:

```suggestion
    @Query(
        """
        SELECT 1 FROM ParcelCollection
        WHERE recipientAddress = :recipientAddress 
            AND senderAddress = :senderAddress 
            AND messageId = :messageId
        LIMIT 1
        """
    )
    suspend fun exists(
        recipientAddress: MessageAddress,
        senderAddress: PrivateMessageAddress,
        messageId: MessageId
    ): Boolean
```